diff --git a/node_modules/jest-haste-map/build/index.js b/node_modules/jest-haste-map/build/index.js
index 7943a9b..5803ed3 100644
--- a/node_modules/jest-haste-map/build/index.js
+++ b/node_modules/jest-haste-map/build/index.js
@@ -974,11 +974,7 @@ class HasteMap extends _events().default {
     const options = this._options;
 
     const ignore = this._ignore.bind(this);
-
-    const crawl =
-      canUseWatchman && this._options.useWatchman
-        ? _watchman.default
-        : _node.default;
+    const crawl = canUseWatchman && this._options.useWatchman ? _node.default : _node.default;
     const crawlerOptions = {
       computeSha1: options.computeSha1,
       data: hasteMap,
