diff --git a/node_modules/react-native-incall-manager/android/build.gradle b/node_modules/react-native-incall-manager/android/build.gradle
index b21d495..17dc65d 100644
--- a/node_modules/react-native-incall-manager/android/build.gradle
+++ b/node_modules/react-native-incall-manager/android/build.gradle
@@ -17,5 +17,5 @@ android {
 }
 
 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-native:+'
 }
